### DISCUSSION TITLE
Is hermes not compatible with go relayer?

### DISCUSSION THREAD

**Discussion by ImJeremyHe** [informalsystems/hermes#2771] [Q&A] [answered]
I found that there was an only `rpc_addr` in the config file in [go relayer](https://github.com/cosmos/relayer). And `hermes` requires `rpc_addr, grpc_addr, ws_addr`.

**Comment by romac** [comment, marked as answer]
What do you mean by "compatible"?

Regardless, Hermes and the Go relayer use a different way of gathering data from the chain. At the moment, the Go relayer only relies on the RPC endpoints whereas Hermes uses both RPC, gRPC and the WebSocket stream.

**Reply by ImJeremyHe** [reply]
Is `hermes` only used as a relayer for chains written in Rust(tendermint-rs and ibc-rs)?

**Reply by adizere** [reply]
No. The language in which the relayer is written (Go, Rust) does not imply anything about the type of networks that relayer can operate on. Both Hermes and go-relayer can operate on both Go and Rust-based IBC networks.
